src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java

```diff
@@ -171,25 +202,26 @@ private BlockPos fillProxyList(@NotNull BlockPos target, double distanceToPath)
     }
 
     /**
+     * Reset the proxy.
+     */
+    public void reset()
+    {
+        currentProxy = null;
+        proxyList = new ArrayList<>();
+    }
+
+    /**
      * Returns a proxy point to the goal for the miner especially.
      *
      * @param target         the target.
      * @param distanceToPath the total distance.
      * @return a proxy or, if not applicable null.
      */
     @NotNull
-    private BlockPos getMinerProxy(final BlockPos target, final double distanceToPath)
+    private BlockPos getMinerProxy(final BlockPos target, final double distanceToPath, @NotNull final BuildingMiner building)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "getMinerProxy" is 27 which is greater than 15 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 60 to the 20 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)